admin/src/services/api.js

```diff
@@ -239,7 +243,11 @@ class api {
 
   uploadID(youngId, file, metadata = {}) {
     let formData = new FormData();
-    const safeFilename = encodeURIComponent(file.name.replace(/'/g, ""));
+    const safeFilename = encodeURIComponent(file.name.replaceAll(/'/g, ""));
```

Comment:
function safeFileName(originalName) {
    // Replace problematic characters with underscore
    let safeName = originalName.replace(/['\\/:*?"<>|\x00-\x1F\x80-\x9F]/g, "_");

    // Trim leading and trailing whitespaces
    safeName = safeName.trim();

    // URI encode the file name to ensure it's safe for transport
    safeName = encodeURIComponent(safeName);

    return safeName;
}

function uploadFile(path, files) {
    let formData = new FormData();

    for (let file of files) {
        file.name = safeFileName(file.name);
        formData.append(file.name, file);
    }
}

Use this instead

Reply:
Check how we use the key "file.name" from the back. Is it mandatory ?